### ISSUE TITLE
Techguns compatibility

### ISSUE THREAD

**Issue by aweoneitor2001** [brandon3055/Draconic-Evolution#1167]
Techguns' weapons can bypass draconic and wyvern armor's shield in 1.12.2

**Comment by brandon3055** [comment]
If thats the case then it is probably deliberate. I could dig through their code and add some countermeasure to DE but i dont really feel like getting into a patch war with some random mod so you should probably talk to them about this.

**Comment by Froggy800** [comment]
pWn3d1337/Techguns2#106
It doesn't appear to be deliberate, so there should be no problem if you release a patch for it. I could be wrong about this, but I think that the issue exists because Techguns uses different types of damage, so it cannot be fixed from their side.

**Comment by triffman33** [comment]
I would be very appreciative if this was looked into again @brandon3055 since it seems to be just an accidental bypass with different damage types 😄

**Comment by brandon3055** [comment]
Hmm... Its hard to imagine a mod "accidentally bypassing" the shield considering it works by blocking damage events... But i will look into it.

**Comment by triffman33** [comment]
It may be because they added a few damage types https://imgur.com/a/JYrwWQG
But nevertheless thank you for taking the time to look into it :)

**Comment by TerdyTheTerd** [comment]
TechGuns utilizes its own DamageSystem class that directly calls the Minecraft EntityLivingBase.damageEntity (func_70665_d) after having cancelled the OnLivingAttack event (if the source is an instance of TGDamageSource), which then never triggers forges event hook not allowing DE from doing its thing. I solved this in my fork of the mod by calculating the final damage then applying it like normally, allowing DE to then do its thing, but I have yet to make a PR with it.

TL:DR; Techguns issue, cancels forge event so DE can't do its thing

**Comment by B-Knight49** [comment]
Just thought I'd 'bump' this to see if any progress has been made.

@TerdyTheTerd - I noticed you mentioned a third-party/custom compatibility mod/patch over on the TechGuns github page. I extend the same question; any progress?

**Comment by TerdyTheTerd** [comment]
@B-Knight49 I placed my changes in a fork and notified the users requesting the patch from TG. None of them ever replied confirming the patch to work, so I never put any more time into it. TG was also still migrating and implementing their damage system, so my patch was only a partial one (TG turrets and custom armor do not work properly, only the weapons). Unless TG finishes their damage system I will not be working further on a patch.

**Comment by B-Knight49** [comment]
If I had to guess, I'd assume that the people who requested the patch aren't aware of how to compile the source code into a jar file. It took me 40 minutes just to get Gradle working properly because it didn't want to play nice - but I digress. 

Your DE-Compat branch works as you say. Thanks for the heads up, hopefully it gets an official patch soon.

**Comment by TerdyTheTerd** [comment]
Ahh that could be part of it, I'll keep an eye out for any commits that finalize the damage system so I can make a proper PR.

**Comment by MasterZero** [comment]
I've build current version of techguns  #(https://github.com/pWn3d1337/Techguns2/commit/4e26523cb0556ad56fefbee83c4df3fe94a426be) with your changes @TerdyTheTerd. The people who requested the patch can find it on my fork

**Comment by MasterZero** [comment]
I have better solution for damage system so it will handle damage as it should in both mods. Please, visit my [repo](https://github.com/MasterZero/Techguns2/tree/2.0.2.0_pre3.2_de_compat) who is still searching for solution.

**Comment by spiro-angelakis** [comment]
> I have better solution for damage system so it will handle damage as it should in both mods. Please, visit my [repo](https://github.com/MasterZero/Techguns2/releases/tag/2.0.2.0_pre3.2_de_compat) who is still searching for solution.

I will be trying this out on the next season (13) of SNG on our server, if it works I greatly appreciate you doing the time to make it. My players are too fond of both DE and TG to remove either of them.

Also, props to brandon for making such a unique mod.

Here's our website if y'all want to check out our modpack and server.
https://www.stevesnewgalaxy.com/
